onnxruntime/core/providers/qnn/builder/qnn_utils.cc

```diff
@@ -319,6 +319,8 @@ std::ostream& operator<<(std::ostream& out, const Qnn_Tensor_t& tensor) {
   }
   out << ")";
   out << " memType=" << GetQnnTensorMemType(tensor);
+// TODO: the code below has compilation errors with the latest ABSL
```

Comment:
+@hectorsvc @adrianlizarraga FYI , is this ok?

Reply:
The build error was:
```
In file included from /usr/include/c++/11/random:49,
                 from /mnt/vss/_work/1/s/build/Release/_deps/abseil_cpp-src/absl/algorithm/container.h:47,
                 from /mnt/vss/_work/1/s/build/Release/_deps/abseil_cpp-src/absl/container/flat_hash_set.h:40,
                 from /mnt/vss/_work/1/s/include/onnxruntime/core/common/inlined_containers.h:21,
                 from /mnt/vss/_work/1/s/include/onnxruntime/core/graph/graph.h:23,
                 from /mnt/vss/_work/1/s/onnxruntime/core/framework/node_unit.h:15,
                 from /mnt/vss/_work/1/s/onnxruntime/core/providers/qnn/builder/qnn_utils.h:13,
                 from /mnt/vss/_work/1/s/onnxruntime/core/providers/qnn/builder/qnn_utils.cc:12:
/usr/include/c++/11/bits/random.h: In instantiation of ‘class std::discrete_distribution<float>’:
/mnt/vss/_work/1/s/onnxruntime/core/providers/qnn/builder/qnn_utils.cc:324:25:   required from here
/usr/include/c++/11/bits/random.h:5281:49: error: static assertion failed: result_type must be an integral type
 5281 |       static_assert(std::is_integral<_IntType>::value,
      |                                                 ^~~~~
/usr/include/c++/11/bits/random.h:5281:49: note: ‘std::integral_constant<bool, false>::value’ evaluates to false
/usr/include/c++/11/bits/random.h: In instantiation of ‘class std::binomial_distribution<float>’:

```

And the build command was:
/usr/bin/c++ -DCPUINFO_SUPPORTED_PLATFORM=1 -DEIGEN_MPL2_ONLY -DEIGEN_USE_THREADS -DNSYNC_ATOMIC_CPP11 -DONLY_C_LOCALE=0 -DONNX_ML=1 -DONNX_NAMESPACE=onnx -DONNX_USE_LITE_PROTO=1 -DORT_ENABLE_STREAM -DPLATFORM_POSIX -DUSE_QNN=1 -D_GNU_SOURCE -D__ONNX_NO_DOC_STRINGS -I/mnt/vss/_work/1/s/build/Release/_deps/utf8_range-src -I/mnt/vss/_work/1/s/include/onnxruntime -I/mnt/vss/_work/1/s/include/onnxruntime/core/session -I/mnt/vss/_work/1/s/build/Release/_deps/pytorch_cpuinfo-src/include -I/mnt/vss/_work/1/s/build/Release/_deps/google_nsync-src/public -I/mnt/vss/_work/1/s/build/Release -I/mnt/vss/_work/1/s/onnxruntime -I/mnt/vss/_work/1/s/build/Release/_deps/abseil_cpp-src -I/mnt/vss/_work/1/s/build/Release/_deps/safeint-src -I/mnt/vss/_work/1/s/build/Release/_deps/gsl-src/include -I/mnt/vss/_work/1/s/build/Release/_deps/date-src/include -I/mnt/vss/_work/1/s/build/Release/_deps/onnx-src -I/mnt/vss/_work/1/s/build/Release/_deps/onnx-build -I/mnt/vss/_work/1/s/build/Release/_deps/protobuf-src/src -I/mnt/vss/_work/1/s/build/Release/_deps/flatbuffers-src/include -I/mnt/vss/_work/1/s/build/Release/_deps/mp11-src/include -I/mnt/vss/_work/_temp/qnn-v2.23.0.240531/include/QNN -I/mnt/vss/_work/_temp/qnn-v2.23.0.240531/include -DNDEBUG -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -O3 -pipe -fstack-clash-protection -fcf-protection -ffunction-sections -fdata-sections -Wno-restrict  -DCPUINFO_SUPPORTED -O3 -DNDEBUG -std=gnu++17 -fPIC -Wall -Wextra -Wno-deprecated-copy -Wno-nonnull-compare -Wno-unknown-pragmas -Werror -MD -MT CMakeFiles/onnxruntime_providers_qnn.dir/mnt/vss/_work/1/s/onnxruntime/core/providers/qnn/builder/qnn_utils.cc.o -MF CMakeFiles/onnxruntime_providers_qnn.dir/mnt/vss/_work/1/s/onnxruntime/core/providers/qnn/builder/qnn_utils.cc.o.d -o CMakeFiles/onnxruntime_providers_qnn.dir/mnt/vss/_work/1/s/onnxruntime/core/providers/qnn/builder/qnn_utils.cc.o -c /mnt/vss/_work/1/s/onnxruntime/core/providers/qnn/builder/qnn_utils.cc

Reply:
It's OK, the code here just prints information for debugging purposes.

Reply:
we should try to fix it later.